src/Xamarin.Android.Tools.ApiXmlAdjuster/JavaApi.XmlModel.cs

```diff
@@ -309,7 +315,7 @@ public JavaTypeParameter (JavaTypeParameters parent)
 		
 		public override string ToString ()
 		{
-			return Name + (GenericConstraints == null ? null : GenericConstraints.ToString ());
+			return Name;
```

Comment:
Basically we have right to choose what to output in this ToString() overrides.

There are places that should output generic constraints and that should NOT. When writing generic type parameter definitions they should be there. When writing USE of generic type parameters they should NOT. And this generic constraints part were always written.

An exceptional case that it SHOULD be written for USE of generics is that the name is "?" which is handled specially.